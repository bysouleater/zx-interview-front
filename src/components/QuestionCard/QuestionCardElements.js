import styled from 'styled-components';
import { Form } from 'semantic-ui-react';

export const QuestionText = styled.label`
  && {
    font-weight: normal !important;
    font-size: 1.1em !important;
  }
`;

export const QuestionInput = styled(Form.Input)`
  .ui.input input {
    border-top: 0 !important;
    border-left: 0 !important;
    border-right: 0 !important;
    border-radius: 0 !important;
  }
`;
