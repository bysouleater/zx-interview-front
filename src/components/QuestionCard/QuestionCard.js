import PropTypes from 'prop-types';
import React from 'react';
import { Form, Segment } from 'semantic-ui-react';
import { QuestionInput, QuestionText } from './QuestionCardElements';

const QuestionCard = ({ question, setAnswerValue }) => (
  <Segment>
    <Form>
      <Form.Field required={question.required}>
        <QuestionText>{question.question}</QuestionText>
        <QuestionInput
          placeholder="Your answer"
          type={question.type}
          error={question.hasError}
          value={question.answer || ''}
          onChange={(e, data) => setAnswerValue(question.id, data.value)}
        />
      </Form.Field>
    </Form>
  </Segment>
);

QuestionCard.propTypes = {
  question: PropTypes.shape({
    id: PropTypes.string,
    question: PropTypes.string,
    required: PropTypes.bool,
    type: PropTypes.oneOf(['text', 'number', 'date']),
    hasError: PropTypes.bool,
    answer: PropTypes.string,
  }).isRequired,
  setAnswerValue: PropTypes.func.isRequired,
};

export default QuestionCard;
