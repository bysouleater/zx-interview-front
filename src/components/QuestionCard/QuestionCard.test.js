import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import QuestionCard from './QuestionCard';


test('renders question in initial state', () => {
  const question = {
    id: '1',
    question: 'Write a lambda function to fetch data from a DynamoDB and a relational DB. Upload your code in your GitHub account and share URL of the repository to validate the code. Try to use dynamic configurations and using Serverless framework is mandatory.',
    required: true,
    type: 'text',
  };
  const { container } = render(<QuestionCard question={question} setAnswerValue={() => {}} />);
  expect(container).toMatchSnapshot();
});

test('renders question with value and error', () => {
  const question = {
    id: '1',
    question: 'Write a lambda function to fetch data from a DynamoDB and a relational DB. Upload your code in your GitHub account and share URL of the repository to validate the code. Try to use dynamic configurations and using Serverless framework is mandatory.',
    required: true,
    type: 'text',
    answer: 'something',
    hasError: true,
  };
  const { container } = render(<QuestionCard question={question} setAnswerValue={() => {}} />);
  expect(container).toMatchSnapshot();
});

test('calls update value on change event', () => {
  const question = {
    id: '1',
    question: 'Write a lambda function to fetch data from a DynamoDB and a relational DB. Upload your code in your GitHub account and share URL of the repository to validate the code. Try to use dynamic configurations and using Serverless framework is mandatory.',
    required: true,
    type: 'text',
  };
  const setAnswerValue = jest.fn();

  const { container } = render(<QuestionCard
    question={question}
    setAnswerValue={setAnswerValue}
  />);

  fireEvent.change(container.querySelector('input'), { target: { value: 'a' } });

  expect(setAnswerValue).toHaveBeenLastCalledWith(question.id, 'a');
});
