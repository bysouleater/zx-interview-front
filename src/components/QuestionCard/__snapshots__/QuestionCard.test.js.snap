// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders question in initial state 1`] = `
<div>
  <div
    class="ui segment"
  >
    <form
      class="ui form"
    >
      <div
        class="required field"
      >
        <label
          class="sc-AxjAm iLMcqR"
        >
          Write a lambda function to fetch data from a DynamoDB and a relational DB. Upload your code in your GitHub account and share URL of the repository to validate the code. Try to use dynamic configurations and using Serverless framework is mandatory.
        </label>
        <div
          class="field sc-AxirZ gvygHu"
        >
          <div
            class="ui input"
          >
            <input
              placeholder="Your answer"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`renders question with value and error 1`] = `
<div>
  <div
    class="ui segment"
  >
    <form
      class="ui form"
    >
      <div
        class="required field"
      >
        <label
          class="sc-AxjAm iLMcqR"
        >
          Write a lambda function to fetch data from a DynamoDB and a relational DB. Upload your code in your GitHub account and share URL of the repository to validate the code. Try to use dynamic configurations and using Serverless framework is mandatory.
        </label>
        <div
          class="error field sc-AxirZ gvygHu"
        >
          <div
            class="ui input"
          >
            <input
              aria-invalid="true"
              placeholder="Your answer"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
